### ISSUE TITLE
BUG in tools/train.py and Confusion about the method of calculating ACC

### ISSUE THREAD

**Issue by zhongqiu1245** [open-mmlab/mmsegmentation#427]
**Checklist**

1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version.

**Describe the bug**
Hi, dear authors
I found that when using multiple GPUs for training (`dis_train`), no matter how much `gpu_nums` I input, the `gpu_ids` in the `config` is always equal to `range (0, 1)`, and only 1 GPU worked, but mmdet's `dis_train `does not have this problem.

**Reproduction**
1. What command or script did you run?

   `bash tools/dist_train.sh configs/mine/new_deeplabv3plus.py 2 --work-dir my_work_mmseg`

2. Did you make any modifications on the code or config? Did you understand what you have modified?
    Yes
3. What dataset did you use?
      VOC format

**Bug fix**
It should be solved by adding these three lines in the picture after line 96 of the `train.py`
![2021-03-16 15-50-30屏幕截图](https://user-images.githubusercontent.com/48090608/111275127-c5ec7800-8670-11eb-9b3c-b853fbd08ea4.png)
This is exactly the difference between `train.py` in `mmseg` and `train.py` in `mmdet`.

**Confusion about the method of calculating Acc**
I have seen the way to calculate *acc*. I'm curious, isn't the way of calculating *acc* in mmseg is calculating *recall*?(`mmsegmentation/mmseg/core/evaluation/metrics.py`)
![屏幕截图 2021-03-16 160857](https://user-images.githubusercontent.com/48090608/111277326-4d3aeb00-8673-11eb-9063-30f73d5fb0fd.jpg)

total_area_label = Ground Truth = TP + FN
total_area_intersect = TP
total_area_pred_label = TP + FP
total_area_union = TP + FN + FP

Recall = TP / （TP + FN）, but Precision = TP / （TP + FP）
Should it be changed to `acc = total_area_intersect / total_area_pred_label `?
![v2-55e6f657baf2b9a8fd1827f460e1e7b6_hd](https://user-images.githubusercontent.com/48090608/111278647-bd963c00-8674-11eb-9e42-5288cea6fbe8.jpg)

**Comment by xvjiarui** [comment]
Hi. acc here is indeed recall. 

As for the GPU issue, would you mind create a PR to fix it?

**Comment by zhongqiu1245** [comment]
OK, I will create a PR to fix it soon.
Sorry to interrupt you, I have two more questions
1. When calculating false negatives (FP) in loss, is FP = torch.sum(torch.mul(1-pred, target) * valid_mask, dim=1) or FP = torch.sum(torch.mul(1 -pred, target), dim=1)?
2. How do we calculate the number of true negatives (TN) in metrics.py?
Thank you

**Comment by xvjiarui** [comment]
Hi @zhongqiu1245 
1. I think it should be first one. 
2. The `TN` is not calculated explicitly. As simple explanation could be that, we could consider a multi-label pixel classification and multiple single-label pixel classification. We are trying to predicting whether this pixel is class A, B, C ... or not.

**Comment by zhongqiu1245** [comment]
thank you

**Comment by lorinczszabolcs** [comment]
> Hi. acc here is indeed recall.
> 
> As for the GPU issue, would you mind create a PR to fix it?

Hi! Is this the intented / correct way to calculate it? In my mind I think accuracy shouldn't be equal to recall, but I might be wrong. Was this corrected in a subsequent PR, or did I miss it?
